gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -51,9 +70,20 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MysqlDatasetStateStore.class);
 
+  private static final String SELECT_JOB_STATE_LATEST_BY_PREFIX_TEMPLATE =
+          "SELECT state, CASE WHEN locate(?, table_name) = 0 THEN '' "
```

Comment:
I would say that we have seen no performance issues with listing vs accessing via an alias and we have been running largely similar code for over a year.  Many of these concerns were discussed and dismissed when #882 was proposed.  Can you also talk to @sahilTakiar about this?

I feel like including both 'alias-based' and 'time-based' complicates the system and will lead to an explosion on state stores.  I think we should try to reduce options where possible so that the system stays clear and maintainable.  Additionally, I believe that the `alias-based` approach is problematic because creating the state and the alias are not an atomic operation.  Additionally, when using S3 as a state store, updates eventual which causes serious problems because all changes to the `current` file are updates.  The effect the old state is used and data is duplicated.